Lesson-4/src/components/WelcomeModal.js

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+export default class WelcomeModal extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      modal: true
```

Comment:
```suggestion
      modal: false
```

вот теперь есть смысл в `componentDidMount()` :)